third_party_libraries/kml.js

```diff
@@ -501,99 +504,84 @@ L.Util.extend(L.KML, {
 
 	},
 
-
+	/**
+	 * Goes through a placemark kml element and generates a leaflet marker, based
+	 *		based on how many areas each placemark has
+	 *
+	 * @param {kml Element} place A placemark kml element
+	 * @param {xml} xml Xml data related to the placemark
+   * @param {path} style Style options for the placemark
+	 * @return {layer[]} layerArray An array of layers that contain leaflet
+	 * 		markers
+	 */
 	parsePlacemark: function (place, xml, style) {
-		var i, j, el, options = {};
-		el = place.getElementsByTagName('styleUrl');
-
-		for (i = 0; i < el.length; i++) {
-			var url = el[i].childNodes[0].nodeValue;
-			for (var a in style[url])
-			{
-				// for jshint
-				if (true)
-				{
-					options[a] = style[url][a];
-				}
+		var layerArray = [];
+		var areas = this.parseExtendedData(place);
+
+	 	for (area in areas) {
+			var options = this.getOptions(place, style)
+			var layer = this.initializeLayer(place, options, xml);
+			var name = this.parseName(place);
+			var description = this.parseDescription(place);
+
+			if (layer) {
+				layer.bindPopup("<h1>" + name + "</h1><b1>" + description + "</b1>", {offset: new L.Point(0,-20)});
+				layer.options.icon = this.createCustomMarker(areas[area]);
+				layer.options.markerType = areas[area].toLowerCase();
+				layerArray.push(layer);
 			}
 		}
-		var layers = [];
 
-		var parse = ['LineString', 'Polygon', 'Point'];
-		for (j in parse) {
-			// for jshint
-			if (true)
-			{
-				var tag = parse[j];
-				el = place.getElementsByTagName(tag);
-
-				for (i = 0; i < el.length; i++) {
-					var l = this["parse" + tag](el[i], xml, options);
-					if (l) { layers.push(l); }
-				}
-			}
-		}
-
-		if (!layers.length) {
-			return;
-		}
-		var layer = layers[0];
-		if (layers.length > 1) {
-			layer = new L.FeatureGroup(layers);
-		}
-
-		var name, descr = "";
-		el = place.getElementsByTagName('name');
-		if (el.length && el[0].childNodes.length) {
-			name = el[0].childNodes[0].nodeValue;
-		}
+		return layerArray;
+	},
 
-		el = place.getElementsByTagName('description');
+	/**
+	 * Reads the values for the data inside an Extended Data kml tag
+	 *
+	 * @param {kml Element} place A placemark XML element
+	 * @return {string[]} dataValueStrings An array of the strings of values
+	 */
+	parseExtendedData: function (place) {
+		var dataValueStrings = [];
+		var data = place.getElementsByTagName('Data');
+		var dataValues = data[0].children;
+		for (i = 0; i < dataValues.length; i++) {
```

Comment:
Since this is just a [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), would it pay to rewrite it as such and just return the result?